packages/webhooks/lib/sync.ts

```diff
@@ -1,101 +1,114 @@
     syncName: string;
     model: string;
     now: Date | undefined;
-    responseResults?: SyncResult;
-    syncType: SyncType;
+    operation: SyncType;
     error?: ErrorPayload;
+    responseResults?: SyncResult;
+    success: boolean;
     activityLogId: number | null;
     logCtx?: LogContext | undefined;
-}): Promise<void> => {
-    if (!shouldSend(environment, 'sync')) {
+} & ({ success: true; responseResults: SyncResult } | { success: false; error: ErrorPayload })): Promise<void> => {
+    if (!webhookSettings) {
         return;
     }
 
-    const noChanges =
-        responseResults?.added === 0 && responseResults?.updated === 0 && (responseResults.deleted === 0 || responseResults.deleted === undefined);
-
-    if (!environment.always_send_webhook && noChanges) {
-        await logCtx?.info('There were no added, updated, or deleted results. No webhook sent, as per your environment settings');
-
+    if (!shouldSend({ success, type: 'sync', webhookSettings, operation })) {
         return;
     }
 
-    const success = typeof error === 'undefined';
-
     const body: NangoSyncWebhookBody = {
         from: 'nango',
-        type: WebhookType.SYNC,
+        type: 'sync',
         connectionId: connection.connection_id,
         providerConfigKey: connection.provider_config_key,
         syncName,
-        success,
         model,
-        syncType
+        syncType: operation
     };
 
+    let successBody: NangoSyncWebhookBodySuccess = {} as NangoSyncWebhookBodySuccess;
+    let errorBody: NangoSyncWebhookBodyError = {} as NangoSyncWebhookBodyError;
+    let endingMessage = '';
+
     if (success) {
-        body.queryTimeStamp = null;
-        body.modifiedAfter = dayjs(now).toDate().toISOString();
-        body.queryTimeStamp = syncType !== 'INITIAL' ? (now as unknown as string) : null;
-    }
+        const noChanges =
+            responseResults?.added === 0 && responseResults?.updated === 0 && (responseResults.deleted === 0 || responseResults.deleted === undefined);
 
-    // TODO send when the failed sync update is opt in
-    if (!success) {
-        return;
-    }
+        if (!webhookSettings.on_sync_completion_always && noChanges) {
+            await logCtx?.info('There were no added, updated, or deleted results. No webhook sent, as per your environment settings');
 
-    if (responseResults) {
-        body.responseResults = {
-            added: responseResults.added,
-            updated: responseResults.updated,
-            deleted: 0
-        };
-    }
+            return;
+        }
 
-    if (!success && error) {
-        body.error = error;
-        body.startedAt = dayjs(now).toDate().toISOString();
-        body.failedAt = new Date().toISOString();
-    }
+        successBody = {
+            ...body,
+            success: true,
+            responseResults: {
+                added: responseResults.added,
+                updated: responseResults.updated,
+                deleted: 0
```

Comment:
why `deleted: 0` and not `deleted: responseResults.deleted || 0`

Reply:
fair, but i'll leave as is